test/unit/ajax.js

```diff
@@ -837,6 +837,41 @@ QUnit.module( "ajax", {
 		};
 	} );
 
+	ajaxTest( "jQuery.ajax() - do not execute scripts from unsuccessful responses (gh-4250)", 2, function( assert ) {
+		return {
+			dataType: "script",
+			url: url( "missing_script_file.js" ),
```

Comment:
We use `url( "404.txt" )` in a few other places to trigger a 404 so that'd be preferred but here we need more - we need the response to return an executable script & verify it's not executed. Something I did in my attempts that I linked to before: https://github.com/mgol/jquery/commit/0cf386936ad7a6735c283d9e3ea5e690c86127a4.